src/components/TextInputs/OneTimeCodeTextInput.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react'
+import { TextInputProps } from 'react-native'
+import styled from 'styled-components/native'
+import Fonts from '../../res/fonts'
+
+const Container = styled.View`
+  flex-direction: column;
+  width: 280px;
+`
+
+const StyledTextInput = styled.TextInput`
+  font-family: '${Fonts.PTMonoRegular}';
+  font-size: 64;
+  color: #333;
+  letter-spacing: 10;
+  height: 60px;
+  width: 300px;
+`
+
+const UnderlinesContainer = styled.View`
+  flex-direction: row;
+  height: 4px;
+  margin: 10px -10px 0 0;
+`
+
+const UnderlineView = styled.View`
+  flex: 1;
+  background-color: #333;
+  margin-right: 10px;
+`
+
+const MAX_LENGTH = 6
+
+export default class OneTimeCodeTextInput extends Component<TextInputProps> {
+  public render() {
+    return (
+      <Container>
+        <StyledTextInput
+          maxLength={MAX_LENGTH}
+          textContentType={'oneTimeCode'}
+          keyboardType={'number-pad'}
+        />
+        <UnderlinesContainer>
+          {[...Array(MAX_LENGTH).keys()].map(i => (
+            <UnderlineView key={`UnderlineView${i}`} />
+          ))}
+        </UnderlinesContainer>
+      </Container>
+    )
+  }
```

Comment:
Just for curiosity, will we have a prop callback (like an `onCodeChange`) and a prop `code` (the OTC itself)? This way we can report the code and fill it when coming from the text message.

Reply:
@lenoirzamboni Hmm I thought it would automatically fill the code after receiving the text message. That's how it works for iOS but I think that's not true for Android, right? You have to have that BroadcastReceiver listening to it and inserting the code in the input, right?

Reply:
@barbosa Yes, on iOS I think it is exactly as you said. About the listener, do you want to implement it on Blocks? I thought we would leave this part for the component that is using the `OneTimeCodeTextInput` and just pass a prop containing the code like we are doing right now with the `AuthCodeInputScreen` on optimus.

Regarding the `onCodeChange`, I think it will be useful for us when we need to login to the apps (testing purpose),  or if in the future the OTC is received using another way, excluding text message and dynamic link, like e-mail. 

What do you think?

Reply:
Yeah, the listener would be implemented elsewhere. Not on blocks.

I got your point. It makes sense to have both props. I will add them 👍 

Thank you, @lenoirzamboni !

Reply:
@lenoirzamboni looking the old component, now I see the reasoning behind those 2 props. It's not a TextInput per se, but a View containing 6 TextInputs.

The new component's Props extends from `TextInputProps`, making it possible to pass its props down to the only TextInput that we have, reusing `onChangeText` and `value` props that `TextInputProps` already has.

By changing the component to something like this...
```diff
<StyledTextInput
    maxLength={MAX_LENGTH}
    textContentType={'oneTimeCode'}
    keyboardType={'number-pad'}
+   {...this.props}
/>
```

... we would be able to use it as:

```typescript
render() {
  return <OneTimeCodeTextInput value={'123456'} onChangeText={this.someFunc} />
}
```

What do you think? Would that suffice?

Reply:
@barbosa Nice! I think so. Sorry for not noticing this.

Reply:
@lenoirzamboni No worries! I will pass `this.props` down to the inner component then. Thank you!